src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1SpaceBookingManagementDomainEventService.kt

```diff
@@ -0,0 +1,104 @@
+  val domainEventService: DomainEventService,
+  val offenderService: OffenderService,
+  val communityApiClient: CommunityApiClient,
+  @Value("\${url-templates.frontend.application}") private val applicationUrlTemplate: UrlTemplate,
+  @Value("\${arrived-departed-domain-events-disabled}") private val arrivedAndDepartedDomainEventsDisabled: Boolean,
+) {
+
+  fun arrivalRecorded(
+    updatedCas1SpaceBooking: Cas1SpaceBookingEntity,
+    previousExpectedDepartureOn: LocalDate? = null,
+  ) {
+    val domainEventId = UUID.randomUUID()
+
+    val application = updatedCas1SpaceBooking.application
+    val premises = updatedCas1SpaceBooking.premises
+
+    val offenderDetails =
+      when (
+        val offenderDetailsResult =
+          offenderService.getPersonSummaryInfoResults(setOf(updatedCas1SpaceBooking.crn), OffenderService.LimitedAccessStrategy.IgnoreLimitedAccess)
+            .firstOrNull()
+      ) {
+        is PersonSummaryInfoResult.Success.Full -> offenderDetailsResult.summary
+        else -> null
+      }
+
+    val keyWorkerStaffDetailsResult = communityApiClient.getStaffUserDetailsForStaffCode(updatedCas1SpaceBooking.keyWorkerStaffCode!!)
+    val keyWorker = when (keyWorkerStaffDetailsResult) {
+      is ClientResult.Success -> keyWorkerStaffDetailsResult.body
+      is ClientResult.Failure -> keyWorkerStaffDetailsResult.throwException()
+    }
+
+    domainEventService.savePersonArrivedEvent(
+      emit = !arrivedAndDepartedDomainEventsDisabled,
+      domainEvent = DomainEvent(
+        id = domainEventId,
+        applicationId = application.id,
+        crn = updatedCas1SpaceBooking.crn,
+        nomsNumber = offenderDetails?.nomsId,
+        occurredAt = updatedCas1SpaceBooking.actualArrivalDateTime!!,
+        cas1SpaceBookingId = updatedCas1SpaceBooking.id,
+        bookingId = null,
+        data = PersonArrivedEnvelope(
+          id = domainEventId,
+          timestamp = OffsetDateTime.now().toInstant(),
+          eventType = EventType.personArrived,
+          eventDetails = PersonArrived(
+            applicationId = application.id,
+            applicationUrl = applicationUrlTemplate.resolve("id", application.id.toString()),
+            bookingId = updatedCas1SpaceBooking.id,
+            personReference = PersonReference(
+              crn = updatedCas1SpaceBooking.crn,
+              noms = offenderDetails?.nomsId ?: "Unknown NOMS Id",
+            ),
+            deliusEventNumber = application.eventNumber,
+            premises = Premises(
+              id = premises.id,
+              name = premises.name,
+              apCode = premises.apCode,
+              legacyApCode = premises.qCode,
+              localAuthorityAreaName = premises.localAuthorityArea!!.name,
+            ),
+            applicationSubmittedOn = application.submittedAt!!.toLocalDate(),
+            keyWorker = StaffMember(
+              staffCode = updatedCas1SpaceBooking.keyWorkerStaffCode!!,
```

Comment:
really minor comment - can we use keyWorker.staffCode here instead for consistency (i appreciate you've duplicated existing code)